main/src/ca/uwaterloo/flix/language/phase/jvm/GenNamespaces.scala

```diff
@@ -17,14 +17,97 @@
 package ca.uwaterloo.flix.language.phase.jvm
 
 import ca.uwaterloo.flix.api.Flix
-import ca.uwaterloo.flix.language.ast.ExecutableAst.Root
-import ca.uwaterloo.flix.language.ast.Type
+import ca.uwaterloo.flix.language.ast.{ExecutableAst, Type}
+import org.objectweb.asm.ClassWriter
+import org.objectweb.asm.Opcodes._
+import ca.uwaterloo.flix.language.ast.Symbol
 
 object GenNamespaces {
 
   // TODO: Documentation and signature
-  def gen(ts: Set[Type], root: Root)(implicit flix: Flix): Map[JvmName, JvmClass] = {
-    Map.empty // TODO
+  def gen(defns: Map[Symbol.DefnSym, ExecutableAst.Def])(implicit flix: Flix): Map[JvmName, JvmClass] = {
+
+    // Defns grouped by their path
+    val groupedDefns = defns.groupBy{ case (sym, defn) =>
+      sym.prefix
+    }
+
+    // Generate each of the namespaces
+    groupedDefns.map{ case (prefix, defnMap) =>
+      val namespace = genNameSpace(prefix, defnMap.values.toList)
+      namespace.name -> namespace
+    }.toMap
+  }
```

Comment:
Ok, so I definitely want to introduce a `NamespaceInfo` which can takes are of all of the above.

Would the following signature make sense to you:

```
case class NamespaceInfo(ns: List[String], defs: List[Def])
```

?

Reply:
Then I would call this function multiple times each with a single `NamespaceInfo`